sympy/ntheory/continued_fraction.py

```diff
@@ -1,6 +1,65 @@
 from sympy.core.numbers import Integer, Rational
+from sympy.core.compatibility import as_int
+from sympy.utilities.misc import filldedent
 
 
+def continued_fraction(a):
+    """Return the Rational or quadratic irrational for the given list
+    representing the continued fraction representation of the same, or
+    the continued fraction representation of the rational or quadratic
+    irrational.
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.continued_fraction import continued_fraction
+    >>> from sympy import sqrt
+    >>> continued_fraction((1 + 2*sqrt(3))/5)
+    [0, 1, [8, 3, 34, 3]]
+    >>> continued_fraction(_)
+    (1 + 2*sqrt(3))/5
```

Comment:
I find the self-inverting behaviour here a bit odd. What's wrong with using continued_fraction_reduce as the inverse?

Reply:
it's reverted